nav2_amcl/src/amcl_node.cpp

```diff
@@ -873,19 +865,6 @@ AmclNode::publishParticleCloud(const pf_sample_set_t * set)
   cloud_with_weights_msg->header.frame_id = global_frame_id_;
   cloud_with_weights_msg->particles.resize(set->sample_count);
 
-  auto cloud_msg = std::make_unique<geometry_msgs::msg::PoseArray>();
-  cloud_msg->header.stamp = this->now();
-  cloud_msg->header.frame_id = global_frame_id_;
-  cloud_msg->poses.resize(set->sample_count);
-  for (int i = 0; i < set->sample_count; i++) {
-    cloud_msg->poses[i].position.x = set->samples[i].pose.v[0];
-    cloud_msg->poses[i].position.y = set->samples[i].pose.v[1];
-    cloud_msg->poses[i].position.z = 0;
-    cloud_msg->poses[i].orientation = orientationAroundZAxis(set->samples[i].pose.v[2]);
-    cloud_with_weights_msg->particles[i].pose = (*cloud_msg).poses[i];
```

Comment:
You deleted the population of the particle could poses / weights for the one that's keeping too... I thought you tested this in rviz?

Reply:
`cloud_msg` is type `PoseArray`, and `cloud_with_weights_msg` is type `ParticleCloud`. I deleted the former.

I did test in rviz, here's a screenshot:
![particle_cloud](https://user-images.githubusercontent.com/11851445/113074958-01f21200-9207-11eb-8dfa-b51ec20b20f4.png)

Reply:
Then where is the particle cloud in that image?

Reply:
Dumb mistake, fixed it in 
![particle_cloud](https://user-images.githubusercontent.com/11851445/113184451-26de9780-9290-11eb-9211-a162d429a3da.png)
f21bed82180dd3765ea5700f873c366c686e2565